data/collector.go

```diff
@@ -160,6 +171,12 @@ func (c *collector) sleepBasedCapture(started chan struct{}) {
 
 		select {
 		case <-c.cancelCtx.Done():
+			// Dan: It's unclear to me why this handling differs from the above error handling with
```

Comment:
This commentary explains why the test only failed sometimes. I have no intention of committing this. But I would argue this is the reason why the test sometimes passes.